stitches/fx_recepie.py

```diff
@@ -0,0 +1,554 @@
+
+    if util.nrow(num_perms[0]["target_experiment"].unique()) > 1:
+        raise TypeError(
+            f"function permute_stitching_recipes should be applied to separate data frames for each target experiment of interest (multiple target ensemble members for a single target experiment is fine)")
+
+    # max number of permutations per target without repeating across generated
+    # ensemble members.
+    N_data_max = min(num_perms[0]['minNumMatches'])
+
+    if N_matches > N_data_max:
+        # TODO this should be written up as a proper python message statement.
+        print("You have requested more recipes than possible for at least one target trajectories, returning what can")
+
+    # Initialize the number of matches to either 0 or the input read from optional:
+    if type(optional) is str:
+        print('initialize to the read-in: has not been translated')
+    else:
+        recipe_collection = pd.DataFrame()
+
+    for target_id in targets['target_ordered_id'].unique():
+        # subset the target info, the target df contains meta information about the run we
+        # and the number of permutations and such.
+        target = targets.loc[targets["target_ordered_id"] == target_id].copy()
+
+        # initialize a recipes for each target
+        recipes_col_by_target = pd.DataFrame()
+        var = target['target_variable'].unique()[0]
+        exp = target['target_experiment'].unique()[0]
+        mod = target['target_model'].unique()[0]
+        ens = target['target_ensemble'].unique()[0]
+
+        # While the following conditions are met continue to generate new recipes.
+        # 1. While we have fewer matches than requested for the target ensemble_member,
+        #    keep going.
+        # 2. Filter the perm_guide to just the target ensemble member in this loop and
+        #    make sure there are at least num_target_windows of time windows: basically
+        #    make sure there is at least one remaining archive match to draw from for
+        #    each target window in this target ensemble. Note this means the perm_guide
+        #    must be updated at the end of every while loop iteration.
+        if util.nrow(recipes_col_by_target) == 0:
+            condition1 = True
+        elif util.nrow(recipes_col_by_target['stitching_id'].unique()) < N_matches:
+            condition1 = True
+        else:
+            condition1 = False
+
+        perm_rows = util.nrow(
+            perm_guide.loc[(perm_guide['target_variable'] == var) & (perm_guide['target_experiment'] == exp) &
+                           (perm_guide['target_model'] == mod) & (perm_guide['target_ensemble'] == ens)]
+                .copy()
+                .drop_duplicates())
+
+        if perm_rows == num_target_windows:
+            condition2 = True
+        else:
+            condition2 = False
+
+        # The index is used to indicate which number of match is being generated
+        index = 1
+
+        # Run the while loop!
```

Comment:
@abigailsnyder & @crvernon I am running into some problem with this while loop, the first unique recipe is returned & the loop continues to try and while other unique recipes but fails to do so. The pandas data frame that is returned only contains 1 recipe for each target.